File: mini-project-client/src/app/models/routeRequest/route-request.ts
import { RouteTravelMode } from "./route-travel-mode"
import { RoutingPreference } from "./routing-preference"
import { TransitPreferences } from "./transit-preferences"

export interface RouteRequest {
  origin: {placeId: string}
  destination: {placeId: string}
  travelMode: RouteTravelMode
  routingPreference?: RoutingPreference
  departureTime: string | null
  arrivalTime: string | null
  computeAlternativeRoutes: boolean
  units?: string
}

File: mini-project-server/src/main/java/project/mini/batch3/vttp/miniprojectserver/models/routeRequest/RouteRequest.java
package project.mini.batch3.vttp.miniprojectserver.models.routeRequest;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import project.mini.batch3.vttp.miniprojectserver.models.routeRequest.enums.RouteTravelMode;
import project.mini.batch3.vttp.miniprojectserver.models.routeRequest.enums.RoutingPreference;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class RouteRequest {
    
    Waypoint origin;
    Waypoint destination;
    RouteTravelMode travelMode;
    RoutingPreference routingPreference;
    String departureTime;
    String arrivalTime;
    boolean computeAlternativeRoutes;
    String units;
}
